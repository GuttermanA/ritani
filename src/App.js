import React, { Component } from "react";
import { octokit } from "api";
import {
  Search,
  // FollowerList,
  UserInfo
} from "scenes";
import { Error } from "components";
import "./App.css";

const defaultState = {
  username: "guttermana",
  userData: {},
  disabled: false,
  error: {
    status: false,
    code: null,
    message: ""
  }
};

class App extends Component {
  state = defaultState;

  componentDidMount = () => {
    octokit.users
      .getByUsername({
        username: this.state.username
      })
      .then(res => this.setState({ userData: res.data }))
      .catch(error =>
        this.setState({
          ...this.state,
          error: { status: true, code: error.status, message: error.message }
        })
      )
      .finally(() => console.log(this.state));
  };

  handleSubmit = event => {
    event.preventDefault();
    this.setState({ disabled: true });
    octokit.users
      .getByUsername({
        username: this.state.username
      })
      .then(res => this.setState({ userData: res.data }))
      .catch(error =>
        this.setState({
          ...this.state,
          error: { status: true, code: error.status, message: error.message }
        })
      )
      .finally(() => console.log(this.state));
  };

  handleChange = event => {
    const target = event.target;
    const value = target.type === "checkbox" ? target.checked : target.value;
    const name = target.name;

    this.setState(
      {
        ...this.state,
        [name]: value
      },
      () => console.log(this.state)
    );
  };

  render() {
    const { userData, error } = this.state;
    return (
      <div>
        <Search
          handleChange={this.handleChange}
          handleSubmit={this.handleSubmit}
        />
        {error.status && <Error code={error.code} message={error.message} />}
        <UserInfo userData={userData} />
      </div>
    );
  }
}

export default App;
