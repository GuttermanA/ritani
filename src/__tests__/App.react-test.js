import React from "react";
import {
  render,
  cleanup,
  mockUserData,
  mockFollowerData
} from "utils/test-utils";
import "@testing-library/jest-dom/extend-expect";
import App from "../App";

const mockState = {
  username: "",
  userData: {},
  followers: {
    data: [],
    currentPage: 1,
    links: {}
  },
  disabled: false,
  error: {
    status: false,
    code: null,
    message: ""
  }
};

describe("<App/>", () => {
  afterEach(cleanup);
  it("renders correctly", () => {
    const component = render(<App />);
    component.state = mockState;
    console.log(component.queryByTestId("app"));
    expect(component).toMatchSnapshot();
  });

  it("has a sticky <div>", () => {
    const { queryByTestId } = render(<App userData />);
    const stickyDiv = queryByTestId("sticky-div");
    expect(stickyDiv).toBeTruthy();
    expect(stickyDiv).toContainElement(queryByTestId("search-bar-container"));
  });
});
