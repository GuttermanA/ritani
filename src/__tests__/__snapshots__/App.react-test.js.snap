// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-testid="app"
      >
        <div
          class="sticky-top"
          data-testid="sticky-div"
        >
          <div
            class="menu fluid center container"
            data-testid="search-bar-container"
          >
            <form
              data-testid="search-form"
            >
              <label
                class="form label"
                data-testid="search-label"
                for="username"
              >
                Enter Github Username to Retrieve Followers
              </label>
              <input
                class="form field"
                data-testid="search-field"
                name="username"
                placeholder="Github Username"
                required=""
                type="text"
                value=""
              />
              <button
                class="form button"
                data-testid="search-button"
                type="submit"
              >
                Search
              </button>
            </form>
            <a
              data-testid="github-repo-link"
              href="https://github.com/GuttermanA/ritani"
              rel="noopener noreferrer"
              style="color: white;"
              target="_blank"
            >
              Github Repository
            </a>
          </div>
        </div>
        <div
          class="fluid message container"
        >
          <h1>
             No Followers Yet
          </h1>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-testid="app"
    >
      <div
        class="sticky-top"
        data-testid="sticky-div"
      >
        <div
          class="menu fluid center container"
          data-testid="search-bar-container"
        >
          <form
            data-testid="search-form"
          >
            <label
              class="form label"
              data-testid="search-label"
              for="username"
            >
              Enter Github Username to Retrieve Followers
            </label>
            <input
              class="form field"
              data-testid="search-field"
              name="username"
              placeholder="Github Username"
              required=""
              type="text"
              value=""
            />
            <button
              class="form button"
              data-testid="search-button"
              type="submit"
            >
              Search
            </button>
          </form>
          <a
            data-testid="github-repo-link"
            href="https://github.com/GuttermanA/ritani"
            rel="noopener noreferrer"
            style="color: white;"
            target="_blank"
          >
            Github Repository
          </a>
        </div>
      </div>
      <div
        class="fluid message container"
      >
        <h1>
           No Followers Yet
        </h1>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "state": Object {
    "disabled": false,
    "error": Object {
      "code": null,
      "message": "",
      "status": false,
    },
    "followers": Object {
      "currentPage": 1,
      "data": Array [],
      "links": Object {},
    },
    "userData": Object {},
    "username": "",
  },
  "unmount": [Function],
}
`;
